### PR TITLE
idea with weekdays programs for tuya-moes thermostats

### PR DESCRIPTION
**Pull request by dukobpa3**
Just an idea but maybe we can go with it and improve.

Would be great to make proper control for this kind of scheduling because whole set of available thermostats have it.

Now we have 2 implementations and bot not much useful:
– long string for all values
– bunch of separate fields

### CODE REVIEW

**Comment by Dukobpa3** [comment]
I'm not using HA and any other visual dashboards except z2m so for me this whistles is important :)

File: src/lib/legacy.ts [hunk 1]
```diff
@@ -6531,6 +6531,52 @@ const toZigbee2 = {
...
+            const getNumbers = (input: exposes.Text): number[] => {
+                input.checkPatternMatch();
+
+                const hourTemperature = input.property.split('/');
+                const hourMinute = hourTemperature[0].split(':', 2);
+                const clamp = (value: number, min: number, max: number): number => {
+                    return Math.max(min, Math.min(max, value));
+                };
+
+                return [clamp(parseInt(hourMinute[0]), 0, 23), clamp(parseInt(hourMinute[1]), 0, 59), clamp(parseInt(hourTemperature[1]) * 2, 5, 35)];
```
Comment (Dukobpa3) [inline src/lib/legacy.ts:6558]: I think here is not needed exception. Clamp is more than enough.

File: src/devices/moes.ts [hunk 2]
```diff
@@ -122,7 +118,7 @@ const definitions: DefinitionWithExtend[] = [
             legacy.tz.moes_thermostat_deadzone_temperature,
             legacy.tz.moes_thermostat_max_temperature_limit,
             legacy.tz.moes_thermostat_min_temperature_limit,
-            legacy.tz.moes_thermostat_program_schedule,
+            legacy.tz.moes_thermostat_program_schedule_v2,
```
Comment (Koenkk) [inline src/devices/moes.ts:121]: Let's not call it v2 but update the existing one.

File: src/lib/legacy.ts [hunk 3]
```diff
@@ -6531,6 +6531,52 @@ const toZigbee2 = {
...
+
+            /* Merge modified value into existing state and send all over in one go */
+            const newProgram = {
+                // @ts-expect-error ignore
+                ...meta.state.program,
+                ...value,
+            };
+
+            const getNumbers = (input: exposes.Text): number[] => {
+                input.checkPatternMatch();
```
Comment (Koenkk) [inline src/lib/legacy.ts:6550]: I don't understand what this does, how can `input` be `exposes.Text` here? Isn't it a string?

File: src/devices/moes.ts (hunk 2 above)
Comment (Dukobpa3) [reply src/devices/moes.ts:121]: It is just draft but if you like the idea I'll change both moes/moesS thermostats for using this one with proper namings
(made pr as draft)

File: src/lib/legacy.ts (hunk 3 above)
Comment (Dukobpa3) [reply src/lib/legacy.ts:6550]: aww
Sure. I've just started to dive into it and missed some core concepts. I've thought there is component and not clean result.

But maybe you'll suggest where can I make this check? I wanted to try to incapsulate all info inside component but not sure where this information can be used. As I see a lot of this stuff doing in z2m-frontend?

Comment (Koenkk) [reply src/lib/legacy.ts:6550]: I would you do a simple regex check here instead of ` input.checkPatternMatch();`

Comment (Dukobpa3) [reply src/lib/legacy.ts:6550]: I can but main idea was to improve something and provide kind of better incapsulated solution

**Pull request closed without merging**